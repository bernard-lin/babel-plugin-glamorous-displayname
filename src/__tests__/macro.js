import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

// removes annoying quotes around string-only stuff
expect.addSnapshotSerializer({
  print(val) {
    return val
  },
  test(val) {
    return typeof val === 'string'
  },
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {filename: path.join(__dirname, 'SomeComponent.js')},
  tests: [
    {
      title: 'simple use case',
      code: `
        import glamorous from '../macro'

        const MyComp = glamorous.div()
      `,
    },
    {
      title: 'multiple usages',
      code: `
        import glamorous from '../macro'

        const MyComp = glamorous.div()
        const MyOtherComp = glamorous.span()
      `,
    },
    {
      title: 'different name',
      code: `
        import g from '../macro'

        const MyComp = g.div()
        const MyOtherComp = g.span()
      `,
    },
  ],
})
