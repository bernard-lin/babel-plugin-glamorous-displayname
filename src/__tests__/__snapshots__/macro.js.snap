// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros different name: different name 1`] = `

import g from '../macro'

const MyComp = g.div()
const MyOtherComp = g.span()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';


const MyComp = g.div.withConfig({
  displayName: 'SomeComponent__MyComp'
})();
const MyOtherComp = g.span.withConfig({
  displayName: 'SomeComponent__MyOtherComp'
})();

`;

exports[`macros multiple usages: multiple usages 1`] = `

import glamorous from '../macro'

const MyComp = glamorous.div()
const MyOtherComp = glamorous.span()

      ↓ ↓ ↓ ↓ ↓ ↓

import glamorous from 'glamorous';


const MyComp = glamorous.div.withConfig({
  displayName: 'SomeComponent__MyComp'
})();
const MyOtherComp = glamorous.span.withConfig({
  displayName: 'SomeComponent__MyOtherComp'
})();

`;

exports[`macros simple use case: simple use case 1`] = `

import glamorous from '../macro'

const MyComp = glamorous.div()

      ↓ ↓ ↓ ↓ ↓ ↓

import glamorous from 'glamorous';


const MyComp = glamorous.div.withConfig({
  displayName: 'SomeComponent__MyComp'
})();

`;
